### PR TITLE
Added dependabot; Updated all dependencies

### PR DESCRIPTION
**Pull request by invinciblevenom**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- Added Dependabot and updated all dependencies 

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by XiangRongLin** [comment]
Unless you have tested each and every feature with each and every configuration I would not merge this

**Comment by TobiGr** [comment]
We have a few dependencies, which cannot be upgraded autoamtically. E.g. exoplayer and the AndroidX libs. Even when they are upgradedmanually, somethimes regressions slip though the review process. I'd suggest to reject this PR.
And as you can see, the tests are failing.

File: app/build.gradle [hunk 1]
```diff
@@ -189,47 +189,47 @@ dependencies {
...
     kapt "frankiesardo:icepick-processor:${icepickVersion}"
 
     // HTML parser
-    implementation "org.jsoup:jsoup:1.13.1"
+    implementation "org.jsoup:jsoup:1.14.2"
 
     // HTTP client
     //noinspection GradleDependency --> do not update okhttp to keep supporting Android 4.4 users
-    implementation "com.squareup.okhttp3:okhttp:3.12.13"
+    implementation "com.squareup.okhttp3:okhttp:4.9.1"
```
Comment (TobiGr) [inline app/build.gradle:232]: Nope.

File: build.gradle [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
-    ext.kotlin_version = '1.4.10'
+    ext.kotlin_version = '1.5.21'
     repositories {
         google()
         mavenCentral()
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.2.2'
+        classpath 'com.android.tools.build:gradle:7.0.1'
```
Comment (TobiGr) [inline build.gradle:11]: build is failing because of this

**Review by TobiGr — CHANGES_REQUESTED** [review]
This cannot be merged

**Comment by invinciblevenom** [comment]
I won't be able to test each and every feature with each and every configuration. Will close this PR. Thanks

**Pull request closed without merging**